src/bin/cargo/commands/rustdoc.rs

```diff
@@ -48,20 +54,45 @@ pub fn cli() -> Command {
 
 pub fn exec(config: &mut Config, args: &ArgMatches) -> CliResult {
     let ws = args.workspace(config)?;
+    let output_format = if let Some(output_format) = args._value_of("output-format") {
+        if matches!(output_format, "json") {
+            config
+                .cli_unstable()
+                .fail_if_stable_opt("--output-format", 12103)?;
+        }
+        OutputFormat::from_str(output_format).unwrap()
+    } else {
+        OutputFormat::Html
+    };
+    let open_result = args.flag("open");
+
+    if open_result && matches!(output_format, OutputFormat::Json) {
+        return Err(CliError::new(
+            anyhow::format_err!("`--open` is not allowed with `json` output format."),
+            101,
+        ));
+    }
```

Comment:
If we don't want support opening JSON file, we could have something like below, so that

* The extra `matches!` in line 69 can be removed.
* `output_format` field can be removed from `DocOptions`, since we compute the `open_result` already.

```suggestion

    let json = matches!(output_format, OutputFormat::Json);
    let open_result = {
        let open = args.flag("open");
        if open && json {
            return Err(CliError::new(
                anyhow::format_err!("`--open` is not allowed with `json` output format."),
                101,
            ));
        }
        open && !json
    };
```

Reply:
I'm okay with allowing `--open` for JSON.

Should we open it in a browser just like we do with the HTML output format?

Reply:
yeah I guess that's fine :)